web/src/app/escalation-policies/PolicyStepFormDest.stories.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+import type { Meta, StoryObj } from '@storybook/react'
+import PolicyStepFormDest, { FormValue } from './PolicyStepFormDest'
+import { expect, userEvent, screen, waitFor, within } from '@storybook/test'
+import { handleDefaultConfig, handleExpFlags } from '../storybook/graphql'
+import { HttpResponse, graphql } from 'msw'
+import { useArgs } from '@storybook/preview-api'
+
+const meta = {
+  title: 'escalation-policies/PolicyStepFormDest',
+  component: PolicyStepFormDest,
+  render: function Component(args) {
+    const [, setArgs] = useArgs()
+    const onChange = (newValue: FormValue): void => {
+      if (args.onChange) args.onChange(newValue)
+      setArgs({ value: newValue })
+    }
+    return <PolicyStepFormDest {...args} onChange={onChange} />
+  },
+  tags: ['autodocs'],
+  parameters: {
+    docs: {
+      story: {
+        inline: false,
+        iframeHeight: 400,
+      },
+    },
+    msw: {
+      handlers: [
+        handleDefaultConfig,
+        handleExpFlags('dest-types'),
+        graphql.query('ValidateDestination', ({ variables: vars }) => {
+          return HttpResponse.json({
+            data: {
+              destinationFieldValidate:
+                vars.input.value === 'https://target.com' ||
+                vars.input.value === 'https://target.com2' ||
+                vars.input.value === '+12225558989',
+            },
+          })
+        }),
+
+        graphql.query('DestDisplayInfo', ({ variables: vars }) => {
+          let body = {}
+          if (vars.input.values[0].value === 'https://target2.com') {
+            body = {
+              data: {
+                destinationDisplayInfo: {
+                  text: vars.input.values[0].value,
+                  iconURL: 'builtin://webhook',
+                  iconAltText: 'Webhook',
+                },
+              },
+            }
+          } else {
+            body = {
+              data: {
+                destinationDisplayInfo: {
+                  text: vars.input.values[0].value,
+                  iconURL: 'builtin://phone-voice',
+                  iconAltText: 'Voice Call',
+                },
+              },
+            }
+          }
+          return HttpResponse.json(body)
```

Comment:
nit: I think this would be clearer to just have `return HttpResponse.json({})` in the first part and then not need the `else` to keep it all flat. Or we could do a switch.

In any case, we should probably switch off of `.type` instead of picking up a field value.
```
switch (vars.input.type) {
case "webhook":
  return HttpResponse.json({data: ... })
case "phone":
  return HttpResponse.json({data: ... })
default:

  return HttpResponse.json({error: ... })
}
```